qutip/solver/flimesolve.py

```diff
@@ -0,0 +1,646 @@
+        The system Hamiltonian wrapped in a FloquetBasis object. Choosing a
+        different integrator for the ``floquet_basis`` than for the evolution
+        of the floquet state can improve the performance.
+
+    tlist : np.ndarray
+        List of 2**n times distributed evenly over one period of the
+            Hamiltonian
+
+    taulist: np.ndarray
+        List of number_of_periods*2**n times distributed evenly over the
+            entire duration of Hamiltonian driving
+
+    Hargs : list
+        The time dependence of the Hamiltonian
+
+   c_ops_and_rates : list of :class:`qutip.Qobj`.
+       List of lists of [collapse operator, collapse operator rate] pairs
+
+    options : dict,optional
+        Options for the solver,see :obj:`FMESolver.options` and
+        `Integrator <./classes.html#classes-ode>`_ for a list of all options.
+    """
+
+    name = "flimesolve"
+    _avail_integrators = {}
+    resultclass = FloquetResult
+    solver_options = {
+        "progress_bar": "text",
+        "progress_kwargs": {"chunk_size": 10},
+        "store_final_state": False,
+        "store_states": None,
+        "normalize_output": True,
+        "method": "adams",
+        "store_floquet_states": False,
+
+    }
+
+    def __init__(
+        self,
+        floquet_basis,
+        c_ops_and_rates,
+        Hargs,
+        *,
+        tlist=None,
+        Nt=None,
+        time_sense=0,
+        options=None
+    ):
+        if isinstance(floquet_basis, FloquetBasis):
+            self.floquet_basis = floquet_basis
+        else:
+            raise TypeError("The ``floquet_basis`` must be a FloquetBasis")
+        self.options = options
+        self.Hdim = np.shape(self.floquet_basis.e_quasi)[0]
+
+        c_ops = []
+        c_op_rates = []
+        for c_op, rate in c_ops_and_rates:
+            if not isinstance(c_op, Qobj):
+                raise TypeError("c_ops must be type Qobj")
+            c_ops.append(c_op)
+            c_op_rates.append(rate)
+
+        self._num_collapse = len(c_ops)
+        if not all(
+            isinstance(c_op, Qobj)
+            for c_op in c_ops
+        ):
+            raise TypeError("c_ops must be type Qobj")
+
+        self.dims = len(self.floquet_basis.e_quasi)
+
+        if Nt != None:
+            self.Nt = Nt
+        elif Nt == None:
+            if tlist is not None:
+                if tlist[1] - tlist[0] >= self.floquet_basis.T:
+                    self.Nt = 2**4
+                elif tlist[1] - tlist[0] < self.floquet_basis.T:
+                    dt = tlist[1] - tlist[0]
+                    tlist_zeroed = tlist - tlist[0]
+                    Nt_finder = abs(tlist_zeroed + dt - self.floquet_basis.T)
+                    self.Nt = list(np.where(
+                        Nt_finder == np.amin(Nt_finder))
+                    )[0][0] + 1
+            else:
+                self.Nt = 2**4
+        self.time_sense = time_sense
+
+        RateDic = _floquet_rate_matrix(
```

Comment:
Let's create a method named `_create_rhs` or similar on the FLiMESolver class and calculate the RHS in one function, rather than making some numpy arrays and then converting them to `QObj` outside.

Inside the `_create_rhs` method we can also add the timing for the creation of the RHS to the stats.

See https://github.com/qutip/qutip/blob/2dc5f3f0a6ad2175e6dc015649989125401fc3bf/qutip/solver/heom/bofin_solvers.py#L661 and then https://github.com/qutip/qutip/blob/2dc5f3f0a6ad2175e6dc015649989125401fc3bf/qutip/solver/heom/bofin_solvers.py#L684.